core/pkg/sync/blob/blob_sync.go

```diff
@@ -0,0 +1,138 @@
+package blob
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/sync"
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/gcsblob" // needed to initialize GCS driver
+)
+
+type Sync struct {
+	Bucket      string
+	Object      string
+	BlobURLMux  *blob.URLMux
+	Cron        Cron
+	Logger      *logger.Logger
+	Interval    uint32
+	ready       bool
+	lastUpdated time.Time
+}
+
+// Cron defines the behaviour required of a cron
+type Cron interface {
+	AddFunc(spec string, cmd func()) error
+	Start()
+	Stop()
+}
+
+func (hs *Sync) Init(_ context.Context) error {
+	return nil
+}
+
+func (hs *Sync) IsReady() bool {
+	return hs.ready
+}
+
+func (hs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	hs.Logger.Info(fmt.Sprintf("starting sync from %s/%s with interval %d", hs.Bucket, hs.Object, hs.Interval))
+	_ = hs.Cron.AddFunc(fmt.Sprintf("*/%d * * * *", hs.Interval), func() {
+		err := hs.sync(ctx, dataSync, false)
+		if err != nil {
+			hs.Logger.Warn(fmt.Sprintf("sync failed: %v", err))
+		}
+	})
+	// Initial fetch
+	hs.Logger.Debug(fmt.Sprintf("initial sync of the %s/%s", hs.Bucket, hs.Object))
+	err := hs.sync(ctx, dataSync, false)
+	if err != nil {
+		return err
+	}
+
+	hs.ready = true
+	hs.Cron.Start()
+	<-ctx.Done()
+	hs.Cron.Stop()
+
+	return nil
+}
+
+func (hs *Sync) ReSync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	return hs.sync(ctx, dataSync, true)
+}
+
+func (hs *Sync) sync(ctx context.Context, dataSync chan<- sync.DataSync, skipCheckingModTime bool) error {
+	bucket, err := hs.getBucket(ctx)
```

Comment:
There are a lot of empty checks on `Bucket` and `Object`, but they are not changing once the syncer is built. Would it be better to move the validation into `Init` and open the bucket in `Init` as well?

Reply:
Moved checks to init. I did it in the first place in during sync mimicking the http sync.
I think (but I'm not sure) it's safe to open bucket once for GCS but I don't know if it's the case for other drivers.